ocaml/xapi/taskHelper.ml

```diff
@@ -178,7 +185,7 @@ let status_is_completed task_status =
   task_status = `success || task_status = `failure || task_status = `cancelled
 
 let complete ~__context result =
-  Context.complete_tracing __context ;
+  let@ () = finally_complete_tracing ?error:None ~__context in
```

Comment:
the `?error:None` is redundant here and could be dropped (although it is fine like this too)

Reply:
I am getting a compiler error when I remove it:
```
File "ocaml/xapi/taskHelper.ml", line 195, characters 12-47:
195 |   let@ () = finally_complete_tracing ~__context in
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error (warning 5 [ignored-partial-application]): this function application is partial,
maybe some arguments are missing.
File "ocaml/xapi/taskHelper.ml", line 195, characters 12-47:
195 |   let@ () = finally_complete_tracing ~__context in
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Error: This expression has type ?error:exn * string -> (unit -> 'a) -> 'a
       but an expression was expected of type ('b -> 'c) -> 'd
Had 2 errors, waiting for filesystem changes...
```
that I am not sure how to fix.

Reply:
https://github.com/xapi-project/xen-api/pull/5735/commits/7e819a2032911b16291b2d1ec10ec08f4b0eee60 I dropped the named `__context`. I think is more clear than adding a new unit parameter.